app/Main.hs

```diff
@@ -60,60 +60,58 @@ formatOne ::
   Mode ->
   -- | Configuration
   Config RegionIndices ->
-  -- | File to format or stdin as 'Nothing'
+  -- | File path supplied on the command line
   Maybe FilePath ->
   IO ExitCode
 formatOne CabalDefaultExtensionsOpts {..} mode config mpath =
-  withPrettyOrmoluExceptions (cfgColorMode config) $
```

Comment:
I think the best way to implement the feature would be abstracting what happens in the `Check` case for reading from file (define a helper function for that) and then putting that in the `Check` case for stdin. The stdin vs file cases are sufficiently different to have different sub-`case`s matching modes.

Reply:
So, we would have,

- `ormoluFile` for in-place and STDOUT modes on file
- `ormoluStdin` for STDOUT mode on STDIN
- (new) `ormoluCheck` for check mode on either file or STDIN?

I don't really like the idea of splitting helper functions on two axes at the same time (mode and type of source), but I will do it if you think it's better

Reply:
Let's leave handling of files in all modes remains as it is, and make a helper function out of this fragment:

```haskell
            case diffText originalInput formattedInput inputFile of
              Nothing -> return ExitSuccess
              Just diff -> do
                runTerm (printTextDiff diff) (cfgColorMode config) stderr
                -- 100 is different to all the other exit code that are emitted
                -- either from an 'OrmoluException' or from 'error' and
                -- 'notImplemented'.
                return (ExitFailure 100)
```

Then use that function to handle diffing for both files and stdin. Formatting of `stdin` can be done with an inlined version of `ormoluInline`.